cmd/rookctl/block/map.go

```diff
@@ -205,33 +201,11 @@ func openRBDFile(hasSingleMajor bool, singleMajorPath, path string) (*os.File, e
 	return fd, nil
 }
 
-func findDevicePath(imageName, poolName, rbdSysBusPath string) (string, error) {
-	rbdDevicesPath := filepath.Join(rbdSysBusPath, rbdDevicesDir)
-	files, err := ioutil.ReadDir(rbdDevicesPath)
-	if err != nil {
-		return "", fmt.Errorf("failed to read rbd device dir: %+v", err)
-	}
-
-	for _, idFile := range files {
-		nameContent, err := ioutil.ReadFile(filepath.Join(rbdDevicesPath, idFile.Name(), "name"))
-		if err == nil && imageName == strings.TrimSpace(string(nameContent)) {
-			// the image for the current rbd device matches, now try to match pool
-			poolContent, err := ioutil.ReadFile(filepath.Join(rbdDevicesPath, idFile.Name(), "pool"))
-			if err == nil && poolName == strings.TrimSpace(string(poolContent)) {
-				// match current device matches both image name and pool name, return the device
-				return rbdDevicePathPrefix + idFile.Name(), nil
-			}
-		}
-	}
-
-	return "", fmt.Errorf("failed to find rbd device path for image '%s' in pool '%s'", imageName, poolName)
-}
-
 func waitForDevicePath(imageName, poolName, rbdSysBusPath string, maxRetries, sleepSecs int) (string, error) {
 	retryCount := 0
 	for {
-		devicePath, err := findDevicePath(imageName, poolName, rbdSysBusPath)
-		if err == nil {
+		devicePath, err := sys.FindRBDDevicePath(imageName, poolName, rbdSysBusPath)
+		if devicePath != "" {
```

Comment:
This was done to make the unit tests easier. Otherwise i need to mock `/dev/rbd`. FindRBDDevicePath returns the devicePath in the format of `/dev/rbd/x`. I have added a check in FindRBDDevicePath to ensure that `/dev/rbd/x` exists.  But of course this means on unittests that it will fail. Do you want me to mock `/dev/rbd`? This means passing a RBDDevicePathPrefix to the function for it to use, which i find odd.

Reply:
I will modify the unit tests in order to make the code look at the error. I think it is a better design.